### PR TITLE
Update coordinate method docstrings

### PR DESCRIPTION
**Pull request by Gamenot**
Update some coordinate related `docstring`s to clarify their coordinate space.

### CODE REVIEW

File: smarts/sstudio/types.py [hunk 1]
```diff
@@ -268,9 +268,7 @@ class BoidAgentActor(SocialAgentActor):
 # This function should be re-callable (although caching is up to the implementation).
 # The idea here is that anything in SMARTS that needs to use a RoadMap
 # can call this builder to get or create one as necessary.
-MapBuilder = NewType(
-    "MapBuilder", Callable[[Any], Tuple[Optional[RoadMap], Optional[str]]]
-)
+MapBuilder = Callable[[Any], Tuple[Optional[RoadMap], Optional[str]]]
```
Comment (Gamenot) [inline smarts/sstudio/types.py:271]: It looks like we needed an alias here rather than a new type.

File: smarts/core/utils/math.py [hunk 2]
```diff
@@ -245,21 +245,27 @@ def low_pass_filter(
...
+    counter-clockwise rotation.
+    """
     # +y = 0 rad.
     angle = (radians + math.pi * 0.5) % (2 * math.pi)
     return np.array((math.cos(angle), math.sin(angle)))
 
 
 def vec_to_radians(v) -> float:
-    """Converts a vector to a radian value."""
+    """Converts a vector to a radian value. [0,+y] is 0 rad with counter-clockwise rotation."""
```
Comment (Gamenot) [inline smarts/core/utils/math.py:257]: These are the strings I was intending to update.

File: smarts/core/coordinates.py [hunk 3]
```diff
@@ -247,11 +258,13 @@ class Pose:
     """A pair of position and orientation values."""
 
     # TODO: these should be np.ndarray
-    position: Sequence  # [x, y, z]
-    orientation: Sequence  # [a, b, c, d] -> a + bi + cj + dk = 0
+    position: Union[NamedTuple, Tuple, np.ndarray]  # [x, y, z]
+    orientation: Union[
+        NamedTuple, Tuple, np.ndarray
+    ]  # [a, b, c, d] -> a + bi + cj + dk = 0
```
Comment (Gamenot) [inline smarts/core/coordinates.py:264]: I set these to be a tuple or numpy array.

File: smarts/core/coordinates.py [hunk 4]
```diff
@@ -247,11 +258,13 @@ class Pose:
...
     heading_: Optional[Heading] = None  # cached heading to avoid recomputing
 
-    def __eq__(self, other) -> bool:
+    def __eq__(self, other: Any) -> bool:
         if not isinstance(other, Pose):
             return False
         return (self.position == other.position).all() and (
```
Comment (Gamenot) [inline smarts/core/coordinates.py:270]: This still seems to be faulty, `all` is not an attribute of `Tuple`, I am surprised we have not encountered errors with this.

File: smarts/core/coordinates.py (hunk 3 above)
Comment (liu-allan) [reply smarts/core/coordinates.py:264]: I thought here https://github.com/huawei-noah/SMARTS/pull/1276/files we only want numpy arrays?

File: smarts/core/coordinates.py [hunk 5]
```diff
@@ -163,7 +172,9 @@ class Heading(float):
     def __init__(self, value=...):
         float.__init__(value)
 
-    def __new__(self, x: Union[SupportsFloat, SupportsIndex] = ...) -> "Heading":
+    def __new__(
+        self, x: Union[SupportsFloat, SupportsIndex, Ellipsis.__class__] = ...
```
Comment (RutvikGupta) [inline smarts/core/coordinates.py:176]: What's the need for ` = ...` ?

**Review by RutvikGupta — APPROVED** [review]

File: smarts/core/utils/math.py [hunk 6]
```diff
@@ -245,21 +245,27 @@ def low_pass_filter(
 
 
 def radians_to_vec(radians) -> np.ndarray:
-    """Convert a radian value to a unit directional vector."""
+    """Convert a radian value to a unit directional vector. 0 rad relates to [0, 1] with
+    counter-clockwise rotation.
+    """
```
Comment (Adaickalavan) [inline smarts/core/utils/math.py:250]: What does "0 rad relates to [0, 1] with ... " mean?

**Review by Adaickalavan — APPROVED** [review]

File: smarts/core/utils/math.py (hunk 6 above)
Comment (Gamenot) [reply smarts/core/utils/math.py:250]: It relates to the vector `[0x, 1y]`.

**Pull request merged**